Sources/TSCBasic/Path.swift

```diff
@@ -457,6 +458,79 @@ private struct WindowsPath: Path, Sendable {
         return !path.withCString(encodedAs: UTF16.self, PathIsRelativeW)
     }
 
+    /// When this function returns successfully, the same path string will have had the prefix removed,
+    /// if the prefix was present. If no prefix was present, the string will be unchanged.
+    static func stripPrefix(_ path: String) -> String {
+        return path.withCString(encodedAs: UTF16.self) { cStringPtr in
+            let mutableCStringPtr = UnsafeMutablePointer(mutating: cStringPtr)
+            let result = PathCchStripPrefix(mutableCStringPtr, path.count + 1)
+            if result == S_OK {
+                return String(decodingCString: mutableCStringPtr, as: UTF16.self)
+            }
+            return path
+        }
+    }
+
+    /// Remove a trailing backslash from a path if the following conditions
+    /// are true:
+    ///   * Path is not a root path
+    ///   * Pash has a trailing backslash
+    /// If conditions are not met then the string is returned unchanged.
+    static func removeTrailingBackslash(_ path: String) -> String {
+        return path.withCString(encodedAs: UTF16.self) { cStringPtr in
+            let mutableCStringPtr = UnsafeMutablePointer(mutating: cStringPtr)
+            let result = PathCchRemoveBackslash(mutableCStringPtr, path.count + 1)
+
+            if result == S_OK {
+                return String(decodingCString: mutableCStringPtr, as: UTF16.self)
+            }
+            return path
+        }
+    }
+
+    /// Create a canonicalized path representation for Windows.
+    /// Returns a potentially `\\?\`-prefixed version of the path,
+    /// to ensure long paths greater than MAX_PATH (260) characters are handled correctly.
+    ///
+    /// - seealso: https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation
+    static func canonicalPathRepresentation(_ path: String) throws -> String {
```

Comment:
suggestion: can we add automated tests that verify this function in isolation?

Reply:
note: this is copied verbatim from swift-foundation; I think a comment that that's the case and it generally shouldn't be changed independently, would be better than tests. Really wish we had a package that could share this implementation rather than having to duplicate it everywhere.